include/zephyr/posix/posix_features.h

```diff
@@ -246,113 +246,196 @@
 /* #define _XOPEN_UUCP (-1L) */
 
 /* Maximum values */
+#undef _POSIX_CLOCKRES_MIN
 #define _POSIX_CLOCKRES_MIN (20000000L)
 
 /* Minimum values */
+#undef _POSIX_AIO_LISTIO_MAX
 #define _POSIX_AIO_LISTIO_MAX               (2)
```

Comment:
Redefining macros / constants en masse using `#undef` + `#define` can lead to undefined behavior when code / headers suddenly can be using different value in different parts of code.

If this is needed it indicates that conflicting headers might be picked up from wrong / unexpected places.

I would like a better explanation as to why this is needed.